src/mlpack/tests/activation_functions_test.cpp

```diff
@@ -1063,3 +1113,21 @@ TEST_CASE("GaussianFunctionTest", "[ActivationFunctionsTest]")
   CheckDerivativeCorrect<GaussianFunction>(desiredActivations,
                                            desiredDerivatives);
 }
+
+/**
+ * Basic test of the Softmin function.
+ */
+TEST_CASE("SoftminFunctionTest", "[ActivationFunctionsTest]")
+{
+  const arma::colvec activationData1("1.7 3.6");
+
+  // Hand-calculated values.
+  const arma::colvec desiredActivations("0.86989153 0.13010847");
```

Comment:
:+1: matches pytorch reference
```python
>>> a = torch.tensor([1.7,3.6])
>>> m(a)
__main__:1: UserWarning: Implicit dimension choice for softmin has been deprecated. Change the call to include dim=X as an argument.
tensor([0.8699, 0.1301])
>>> o = m(a)
>>> o.grad_fn(torch.tensor([1.0,0.0]))
tensor([ 0.1132, -0.1132], grad_fn=<SoftmaxBackwardDataBackward>)

```

Reply:
I am still a bit suspicious about the backward function so i will check with some more values and post the results here, also thanks for reviewing this so quickly :)

Reply:
Hey @birm,
with the input of `4.2, 2.4, 7.0, 6.4` and taking the derivative WRT `1.0, 0.0, 1.0, 0.0`
PyTorch produces : `0.1181, -0.1231,  0.0072, -0.0023`
```
>>> a = torch.tensor([4.2, 2.4, 7.0, 6.4],requires_grad=True)
>>> o = m(a)
>>> o
tensor([0.1385, 0.8378, 0.0084, 0.0153], grad_fn=<SoftmaxBackward>)
>>> o.grad_fn(torch.tensor([1.0,0.0,1.0,0.0]))
tensor([ 0.1181, -0.1231,  0.0072, -0.0023],
       grad_fn=<SoftmaxBackwardDataBackward>)
```
Whereas this implementation produces : `-0.1181, 0.1231, -0.0072, 0.0023`
So changing the current backward from : `g = input % (arma::repmat(arma::sum(gy % input), input.n_rows, 1) - gy);`
To this : `g = input % (gy - arma::repmat(arma::sum(gy % input), input.n_rows, 1));`
inverts the minus sign and so we can get the correct output as per PyTorch.
So i will go ahead and update the backward function here.

Reply:
Also i am surprised that this and the `Softmax` backward are the exact same.

Reply:
Please add the new test case here

Reply:
Done.